src/appengine/libs/auth.py

```diff
@@ -178,6 +178,7 @@ def create_session_cookie(id_token, expires_in):
 
 def get_current_request():
   """Get the current request."""
+  # TODO(singharshdeep): Remove webapp.get_request() after flask migration.
```

Comment:
We call setattr() on this. Does this work as expected on flask.request, which does some magic to make flask.request proxy the actual request, or can this lead to inconsistent states?

Reply:
https://werkzeug.palletsprojects.com/en/1.0.x/wrappers/#mutability-and-reusability-of-wrappers mentions that requests are immutable.

Reply:
yes, as we discussed I have used flask.g for all such setattr() and getattr() on requests.